### PR TITLE
Allow different settings for horizontal/vertical split

### PR DESCRIPTION
**Pull request by 70ray**
Add setSplitPercent() function to imageControl for this.
imageControl, textControl, controlBar, imageWidget & textWidget
use different storage keys for h or v layout and return setup
functions to use the key to find and apply settings.
Switching h/v fires a callback to do this.
control bar names improved to compassPoint etc.

### CODE REVIEW

**Comment by srjfoo** [comment]
@70ray, is there a sandbox available for this?

**Comment by 70ray** [comment]
Sorry forgot to mention Sandbox at: https://www.pgdp.org/~rp31/c.branch/hv_persist

**Comment by 70ray** [comment]
Using the Promise shouldn't make any functional difference, just hopefully makes it easier to see how the code works.

**Comment by srjfoo** [comment]
> Using the Promise shouldn't make any functional difference, just hopefully makes it easier to see how the code works.

I think what I need for helping me see how the code works (by reading the code, that is, instead of just testing) is a javascript tutorial that reminds me of what JS I used to know (which wasn't much), and then carries on into the advanced stuff that you're using. That being said, I've exercised the persistence and tried to also exercise the other controls a bit, and other than image size persisting, I don't notice any change in operation, so that's good.

**Comment by 70ray** [comment]
@srjfoo It should be possible to have different settings for everything in horizonal and vertical layouts; so you could have the control bars at the side in horizontal mode and at the top in vertical mode, and different fonts and font sizes as well as different zoom factors.
(I often feel like you described when looking at Chris's code)

**Comment by srjfoo** [comment]
> @srjfoo It should be possible to have different settings for everything in horizonal and vertical layouts; so you could have the control bars at the side in horizontal mode and at the top in vertical mode, and different fonts and font sizes as well as different zoom factors.

And according to my testing, that's what we seem to have.

> (I often feel like you described when looking at Chris's code)

I'm not alone! :grin:

**Review by srjfoo — APPROVED** [review]

File: scripts/control_bar.js [hunk 1]
```diff
@@ -361,6 +371,12 @@ function makeImageWidget(container, storageKey, align = "C") {
     });
 
     return {
+        setup: function (storageKey) {
+            let imageWidgetKey = storageKey + "-imagewidget";
```
Comment (chrismiceli) [inline scripts/control_bar.js:375]: you use const above, so we could use it here too, though this is a minor comment.

File: scripts/page_browse.js [hunk 2]
```diff
@@ -76,42 +55,60 @@ var maketextControl = function(textArea, storageKey) {
...
     let wrapCheck = $("<input>", {type: 'checkbox'});
 
-    // stored value is "W" or "N", if not set textWrap will be false
-    let textWrap = ("W" === textData.textWrap);
-    wrapCheck.prop("checked", textWrap);
-    setWrap(textWrap);
-
     wrapCheck.change(function () {
-        textWrap = wrapCheck.prop("checked");
+        let textWrap = wrapCheck.prop("checked");
```
Comment (chrismiceli) [inline scripts/page_browse.js:59]: same with const since we don't modify textData.

**Review by chrismiceli — APPROVED** [review]

File: scripts/control_bar.js (hunk 1 above)
Comment (70ray) [reply scripts/control_bar.js]: Done this.

**Pull request merged**